src/wovnio/wovnphp/Url.php

```diff
@@ -38,16 +38,21 @@ public static function addLangCode($uri, $store, $lang, $headers)
 
         $new_uri = $uri;
         $pattern = $store->settings['url_pattern_name'];
+        $site_prefix_path = $store->settings['site_prefix_path'];
         $lang_code = $store->convertToCustomLangCode($lang);
         $lang_param_name = $store->settings['lang_param_name'];
 
-        $no_lang_uri = self::removeLangCode($uri, $pattern, $lang_code, $lang_param_name);
-        $no_lang_host = self::removeLangCode($headers->host, $pattern, $lang_code, $lang_param_name);
+        if ($pattern === 'path' && !empty($site_prefix_path) && !self::matchSitePrefixPath($uri, $site_prefix_path)) {
```

Comment:
This looks like a duplicate of `shouldIgnoreBySitePrefixPath`

Reply:
`shouldIgnoreBySitePrefixPath` using `matchSitePrefixPath` inside.

Reply:
Not sure what you mean. This is exactly the same as what `shouldIgnoreBySitePrefixPath` is doing. It would be better if we made one of these functions private.

Reply:
Actually, shouldn't this be calling `isIgnoredPath`?

Reply:
I agree with using `isIgnoredPath` is better to not modify ignored URL.
But this method is used everywhere.
I don't want to fix it this time because of fear of bad effects.

Reply:
`isIgnoredPath` is only used once in wovn_interceptor.php. I don't see a problem with using it.

Reply:
I think it is actually a bug if we don't use it. If the server redirects to `/ignored_path/`, we should not add lang code to it.

Reply:
Yes. That's right.
I will replace it.

Reply:
Done